server/src/main/java/org/apache/iotdb/db/qp/logical/crud/QueryOperator.java

```diff
@@ -338,15 +338,16 @@ private void convertSpecialClauseValues(QueryPlan queryPlan) {
     }
   }
 
-  private void convertSpecialClauseValues(QueryPlan queryPlan, List<String> measurements)
+  private List<String> convertSpecialClauseValues(QueryPlan queryPlan, List<String> measurements)
       throws QueryProcessException {
     convertSpecialClauseValues(queryPlan);
     // sLimit trim on the measurementColumnList
     if (specialClauseComponent.hasSlimit()) {
       int seriesSLimit = specialClauseComponent.getSeriesLimit();
       int seriesOffset = specialClauseComponent.getSeriesOffset();
-      slimitTrimColumn(measurements, seriesSLimit, seriesOffset);
+      return slimitTrimColumn(measurements, seriesSLimit, seriesOffset);
```

Comment:
Hi, I think just reassigning the `measurements` in the `slimitTrimColumn` method is ok.
```suggestion
      measurements = slimitTrimColumn(measurements, seriesSLimit, seriesOffset);
```